project_key/models/project_project.py

```diff
@@ -0,0 +1,183 @@
+        for rec in self:
+            if rec.name:
+                rec.key = self.generate_project_key(rec.name)
+            else:
+                rec.key = ''
+
+    @api.model
+    def create(self, vals):
+        if 'key' not in vals:
+            if vals['name']:
+                vals['key'] = self.generate_project_key(vals['name'])
+            else:
+                vals['key'] = ''
+
+        new_project = super(ProjectProject, self).create(vals)
+        new_project.create_sequence()
+
+        return new_project
+
+    @api.multi
+    def write(self, values):
+
+        reindex = False
+
+        if 'key' in values:
+            key = values['key']
+            reindex = self.key != key
+
+        res = super(ProjectProject, self).write(values)
+
+        if reindex:
+            # Here we don't expect to have more than one record
+            # because we can not have multiple projects with same KEY.
+            self.update_sequence()
+            self._reindex_task_keys()
+
+        return res
+
+    @api.multi
+    def unlink(self):
+        for project in self:
+            project.task_key_sequence_id.sudo().unlink()
+        return super(ProjectProject, self).unlink()
+
+    @api.model
+    def name_search(self, name, args=None, operator='ilike', limit=100):
+        args = args or []
+        domain = []
+        if name:
+            domain = [
+                '|', ('key', 'ilike', name + '%'), ('name', operator, name)
+            ]
+            if operator in expression.NEGATIVE_TERM_OPERATORS:
+                domain = ['&'] + domain
+        projects = self.search(domain + args, limit=limit)
+        return projects.name_get()
+
+    def create_sequence(self):
+        """
+        This method creates ir.sequence fot the current project
+        :return: Returns create sequence
+        """
+        self.ensure_one()
+        sequence_data = self._prepare_sequence_data()
+        sequence = self.env['ir.sequence'].sudo().create(sequence_data)
+        self.write({'task_key_sequence_id': sequence.id})
+        return sequence
+
+    def update_sequence(self):
+        """
+        This method updates existing task sequence
+        :return:
+        """
+        sequence_data = self._prepare_sequence_data(init=False)
+        self.task_key_sequence_id.sudo().write(sequence_data)
+
+    def _prepare_sequence_data(self, init=True):
+        """
+        This method prepares data for create/update_sequence methods
+        :param init: Set to False in case you don't want to set initial values
+        for number_increment and number_next_actual
+        """
+        values = {
+            'name': "%s %s" % (
+                _("Project task sequence for project "), self.name
+            ),
+            'implementation': 'standard',
+            'code': 'project.task.key.%s' % (self.id,),
+            'prefix': "%s-" % (self.key,),
+            'use_date_range': False,
+            'active': False,
```

Comment:
Why is it set to inactive?

Reply:
At first I didn't want this sequence to be seen as it's maintained by the module itself but now when you mentioning that I've removed that :)